gpflow/kernels/periodic.py

```diff
@@ -65,20 +66,15 @@ def K_diag(self, X: tf.Tensor) -> tf.Tensor:
         return self.base_kernel.K_diag(X)
 
     def K(self, X: tf.Tensor, X2: Optional[tf.Tensor] = None) -> tf.Tensor:
-        if X2 is None:
-            X2 = X
-
-        # Introduce dummy dimension so we can use broadcasting
-        # TODO: does not support kernel broadcasting
```

Comment:
does it now handle broadcasting correctly? if so, could you activate the corresponding test (tests/gpflow/kernels/test_broadcasting.py)?